test/update.js

```diff
@@ -123,6 +122,28 @@ module.exports = function(idProperty, getEngine) {
       })
     })
 
+    it('should always return a new object', function (done) {
```

Comment:
I would like to see a test that describes problem that we've seen more clearly.

`should return a unreferenced overridden object when override is true`